agent/agent_endpoint.go

```diff
@@ -1248,7 +1398,17 @@ func (s *HTTPServer) agentLocalBlockingQuery(resp http.ResponseWriter, hash stri
 			return curResp, err
 		}
 		// Watch returned false indicating a change was detected, loop and repeat
-		// the callback to load the new value.
+		// the callback to load the new value. If agent sync is paused it means
+		// local state is currently being bulk-edited e.g. config reload. In this
+		// case it's likely that local state just got unloaded and may or may not be
+		// reloaded yet. Wait a short amount of time for Sync to resume to ride out
+		// typical config reloads.
+		if syncPauseCh := s.agent.syncPausedCh(); syncPauseCh != nil {
+			select {
+			case <-syncPauseCh:
+			case <-timeout.C:
+			}
+		}
```

Comment:
This is fine to be in this PR but just want to check: this fixes another outstanding bug right? 

I want to note that since we should put it into the CHANGELOG.

Reply:
Kinda. Before this PR the only blocking endpoint was managed proxy config one. It's possible that it could have the same issue when reloading config but I doubt anyone noticed or was affected as the proxies would have just retried again after a second and been fine.

Worst case someone manually fiddling with proxy config endpoint might have observed this behaviour or asked about the spurious error in managed proxy logs but I don't think I've seen it come up yet.